scripts/common.sh

```diff
@@ -193,16 +193,56 @@ function msg_on_error
     fi
 }
 
+
 function diff_values
 {
     (( $# >= 2 )) || fail "diff_values requires at least 2 arguments."
 
     local value1="$1"
     local value2="$2"
-    shift
-    shift
+    shift 2
 
-    diff --unified=0 <(echo "$value1") <(echo "$value2") "$@"
+    if ! diff --unified=0 <(echo "$value1") <(echo "$value2") "$@"
+    then
+        if [ "${DIFFVIEW:-}" == "" ]
+        then
+            printError "ERROR: values differ:"
+            printError "Expected:"
+            printError "${value1}"
+            printError "Obtained:"
+            printError "${value2}"
+        else
+            # Use user supplied diff view binary
+            printError "ERROR: values differ."
+            "$DIFFVIEW" "${file1}" "${file2}"
```

Comment:
We don't have files here. This function is for diffing values stored in variables.